test/apps/iodemo/io_demo.cc

```diff
@@ -948,6 +948,21 @@ class P2pDemoCommon : public UcxContext {
         validate(conn, msg, iomsg_size);
     }
 
+    static inline void check_conn_id(const iomsg_t *msg, UcxConnection *conn)
+    {
+#ifndef NDEBUG
+        if (conn->use_am() && (msg->op == IO_READ_COMP)) {
+            assert(conn->id() == msg->conn_id);
+        } else if (conn->use_am()) {
```

Comment:
lets handle `use_am()` case separately to avoid double checking

Reply:
done